go/vt/tabletserver/query_splitter.go

```diff
@@ -108,27 +114,27 @@ func (qs *QuerySplitter) split(pkMinMax *mproto.QueryResult) ([]proto.QuerySplit
 		}
 		splits = append(splits, *split)
 	} else {
+		boundaries = append(boundaries, sqltypes.Value{})
+		where := qs.sel.Where
 		// Loop through the boundaries and generated modified where clauses
 		start := sqltypes.Value{}
-		clauses := []*sqlparser.Where{}
 		for _, end := range boundaries {
-			clauses = append(clauses, qs.getWhereClause(start, end))
-			start.Inner = end.Inner
-		}
-		clauses = append(clauses, qs.getWhereClause(start, sqltypes.Value{}))
-		// Generate one split per clause
-		for _, clause := range clauses {
-			sel := qs.sel
-			sel.Where = clause
+			bindVars := make(map[string]interface{}, len(qs.query.BindVariables))
+			for k, v := range qs.query.BindVariables {
+				bindVars[k] = v
+			}
+			qs.sel.Where = qs.getWhereClause(bindVars, start, end)
```

Comment:
This is confusing because the function uses sel.Where to generate a new where clause, which is assigned right back here. It will be more readable if you passed in sel.Where, as input argument.